include/ignition/gazebo/components/Factory.hh

```diff
@@ -63,11 +69,21 @@ namespace components
     {
       return std::make_unique<ComponentTypeT>();
     }
+
+    public: virtual std::unique_ptr<BaseComponent> Create(
+                const components::BaseComponent *_data) const override
+    {
+      ComponentTypeT comp(*static_cast<const ComponentTypeT *>(_data));
```

Comment:
> Can't the caller themselves directly create a unique pointer with the component, instead of creating a raw pointer and calling this function to copy it into a unique pointer?

Technically, yes, but this is where the method is used: https://github.com/ignitionrobotics/ign-gazebo/blob/f0a0fe85029fe448b1c12ca522db8ce6f67dc71d/include/ignition/gazebo/components/Factory.hh#L270-L286

`Factory::New` is not a templated method, so I believe that we need to create the unique pointer in `ComponentDescriptor::Create` in order to create an instance of the derived component class, since the `ComponentDescriptor` class is a template class where the class' template parameter is the derived component class. Does that make sense?

Reply:
Ahh gotcha, makes sense. Thanks.